### PR TITLE
Status bar is no longer full-width

### PR DESCRIPTION
**Pull request by VictorVow**
## About The Pull Request

Removes the hideous full-width white status bar and replaces it with a new status bar that:
- Doesn't take all of the screen's width
- Nice and dark
- Hides when there is no status text

[Demo](https://cdn.discordapp.com/attachments/484171634169741312/818293328914939904/output_00001.mp4)

## Why It's Good For The Game

Previous status bar looked horrible (especially on dark mode) and took up way too much space for no reason. Having the extra screen space makes the whole UI look much nicer in addition. 

## Changelog
:cl: stylemistake
tweak: Status bar is no longer full-width, hides when unused and looks nicer. 
/:cl:

Ports:
https://github.com/tgstation/tgstation/pull/57531

### CODE REVIEW

**Comment by ToasterBan** [comment]
it not taking up the whole bottom part makes it look uglier tbh

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1399,3 +1399,9 @@
 /atom/proc/InitializeAIController()
 	if(ai_controller)
 		ai_controller = new ai_controller(src)
+
+//Update the screentip to reflect what we're hovering over
+/atom/MouseEntered(location, control, params)
+	. = ..()
+	// Statusbar
+	status_bar_set_text(usr, name)
```
Comment (Kapu1178) [inline code/game/atoms.dm:1407]: This is going to introduce a delay to using the status bar, god no.

This is also going to be absurdly expensive, every time your mouse fucking MOVES it calls an extra proc for no reason.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by francinum** [comment]
This is so immensely fucking stupid I've crawled out of my hole to call you out.

This is one of the most expensive functions in the game to bind. And you aren't even binding it for good reason.

File: code/game/atoms.dm (hunk 1 above)
Comment (PowerfulBacon) [inline code/game/atoms.dm:1407]: On high pop this is one of the most overtiming functions (Even when doing nothing!!)
Just simply adding /atom/MouseEntered and doing NOTHING with the result is bad. This is why we made an effort to remove it entirely.

Plus its a TG port so I automatically don't like it.

**Comment by PowerfulBacon** [comment]
:key:

**Comment by francinum** [comment]
I know I'm not worth shit anymore, but I expected better.

Even out of you, Victor. Somehow.

🗝️

**Comment by Crossedfall** [comment]
> 🔑

🔑

**Pull request closed without merging**